src/renate/utils/avalanche.py

```diff
@@ -1,55 +1,50 @@
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 from collections import Counter
-from typing import Any, Dict, List, Optional, Tuple, Type
+from typing import Any, Dict, List, Optional, Tuple, Type, Union
 
 import torch
 from avalanche.benchmarks import dataset_benchmark
 from avalanche.core import BasePlugin
 from torch import Tensor
-from torch.utils.data import DataLoader, Dataset, Subset
+from torch.utils.data import DataLoader, Dataset
 
 from renate.data.datasets import _TransformedDataset
+from renate.memory import DataBuffer
+from renate.types import Inputs
 
 
-class AvalancheSubset(Dataset):
-    """Helper class to use dataset subsets with Avalanche.
+class AvalancheDataset(Dataset):
+    """A Dataset consumable by Avalanche updaters."""
 
-    Since Avalanche does not support the Subset class, this class is used to convert subsets.
-    Furthermore, some Avalanche updaters (e.g. iCaRL) directly access the targets attribute.
+    def __init__(self, inputs: Inputs, targets: List[int]):
+        self._inputs = inputs
+        self._targets = targets
+        self.targets = torch.tensor(targets, dtype=torch.long)
```

Comment:
Why is this needed? Is it used by `avalanche`?

Reply:
some avalanche updaters directly access `targets` in the dataset